src/basic/strv.c

```diff
@@ -904,6 +904,22 @@ int strv_extend_n(char ***l, const char *value, size_t n) {
         return -ENOMEM;
 }
 
+int strv_extend_assignment(char ***l, const char *lhs, const char *rhs) {
```

Comment:
this smells a lot like strv_env_assign()...

I wonder if it wouldn't be better to make this new helper more generic, i.e. call this strv_extend_join() that combined strjoin() + strv_extend(). You'd thus call:

```c
r = strv_extend_join(&l, "foo", "=", "bar");
```

I think that would be almost as good to read, less redundant given that strv_env_assign() already exists, and I am sure we can find various other places to use this.